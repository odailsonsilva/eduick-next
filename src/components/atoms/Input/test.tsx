import { fireEvent, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderWithTheme } from '../../../utils/tests/helpers'

import { Input } from '.'

describe('<Input />', () => {
  it('shold render input with label', () => {
    renderWithTheme(<Input label="label" name="label" />)

    expect(screen.getByLabelText('label')).toBeInTheDocument()
  }),
    it('Renders without Label', () => {
      renderWithTheme(<Input />)
      expect(screen.queryByLabelText('Label')).not.toBeInTheDocument()
    }),
    it('Renders with placeholder', () => {
      renderWithTheme(<Input placeholder="placeholder" />)
      expect(screen.getByPlaceholderText('placeholder')).toBeInTheDocument()
    }),
    it('should render input isPassword', () => {
      renderWithTheme(<Input isPassword />)
      expect(screen.getByTestId('button--password')).toBeInTheDocument()
    }),
    it('Changes its value when typing', async () => {
      const onInputChange = jest.fn()

      const { container } = renderWithTheme(
        <Input
          label="label"
          type="text"
          name="label"
          onChange={onInputChange}
        />
      )

      const input = screen.getByRole('textbox')
      fireEvent.change(input, { target: { value: '23' } })
      expect(input.value).toBe('23')

      expect(container.firstChild).toMatchSnapshot()
    })
})
