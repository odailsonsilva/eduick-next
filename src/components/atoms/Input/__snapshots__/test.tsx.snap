// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Changes its value when typing 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #ffffff;
  margin-bottom: 0.5rem;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 5.4rem;
  font-size: 1.6rem;
  color: #ffffff;
  background: transparent;
  border: none;
}

.c2::-webkit-input-placeholder {
  font-size: 1.6rem;
  color: #ffffff;
}

.c2::-moz-placeholder {
  font-size: 1.6rem;
  color: #ffffff;
}

.c2:-ms-input-placeholder {
  font-size: 1.6rem;
  color: #ffffff;
}

.c2::placeholder {
  font-size: 1.6rem;
  color: #ffffff;
}

.c1 {
  border: 1px solid #A68EEF;
  border-radius: 0.4rem;
  height: 5.6rem;
  width: 100%;
  padding: 1.2rem 1.6rem 1.2rem 2.8rem;
  background: #7955E8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 div.wrapper__input__icon {
  margin-top: 4px;
  margin-left: 0.8rem;
  cursor: pointer;
}

@media (max-width:768px) {
  .c1 {
    padding: 1rem 2.2rem 1rem 2.2rem;
  }
}

<div
  class=""
>
  <label
    class="c0"
    for="label"
  >
    label
  </label>
  <div
    class="c1"
  >
    <input
      class="c2"
      id="label"
      name="label"
      type="text"
    />
  </div>
</div>
`;
